modules/c8000v-sdwan/variables.tf

```diff
@@ -0,0 +1,176 @@
+    condition     = length(var.name) >= 2 && length(var.name) <= 50
+    error_message = "Device name should consist of 2 to 50 characters."
+  }
+}
+
+variable "cloud_init_file_id" {
+  description = "C8000V Cloud-Init bootstrap configuration file"
+  type        = string
+  validation {
+    condition     = length(var.cloud_init_file_id) > 0
+    error_message = "Cloud-Init bootstrap is mandatory."
+  }
+}
+
+variable "term_length" {
+  description = "Term length in months"
+  type        = number
+  validation {
+    condition     = can(regex("^(1|12|24|36)$", var.term_length))
+    error_message = "One of following term lengths are available: 1, 12, 24, 36 months."
+  }
+}
+
+variable "notifications" {
+  description = "List of email addresses that will receive device status notifications"
+  type        = list(string)
+  validation {
+    condition     = length(var.notifications) > 0
+    error_message = "Notification list cannot be empty."
+  }
+}
+
+variable "acl_template_id" {
+  description = "Identifier of an ACL template that will be applied on a device."
+  type        = string
+  validation {
+    condition     = length(var.acl_template_id) > 0
+    error_message = "acl template is mandatory."
+  }
+}
+
+variable "additional_bandwidth" {
+  description = "Additional internet bandwidth for a device"
+  type        = number
+  default     = 0
+  validation {
+    condition     = var.additional_bandwidth == 0 || (var.additional_bandwidth >= 25 && var.additional_bandwidth <= 5001)
+    error_message = "Additional internet bandwidth should be between 25 and 2001 Mbps."
+  }
+}
+
+variable "interface_count" {
+  description = "Number of network interfaces on a device. If not specified, default number for a given device type will be used."
+  type        = number
+  default     = 10
+}
+
+variable "vendor_configuration" {
+  description = "Device specific vendor configurations."
+  type = object({
+    hostname        = string
+    siteId          = string
+    systemIpAddress = string
+  })
+
+  validation {
+    condition     = length(var.vendor_configuration.hostname) >= 2 && length(var.vendor_configuration.hostname) <= 10
+    error_message = "Hostname must consist of 2 to 10 characters"
+  }
+
+  validation {
+    condition     = length(var.vendor_configuration.siteId) > 0
+    error_message = "Site ID is mandatory"
+  }
+
+  validation {
+    condition     = length(var.vendor_configuration.systemIpAddress) > 0
+    error_message = "System IP Address is mandatory"
+  }
+}
+
+
+variable "secondary" {
+  description = "Secondary device attributes"
+  type = object({
+    enabled            = bool
+    metro_code         = string
+    cloud_init_file_id = string
+    acl_template_id    = string
+    account_number     = string
+    vendor_configuration = object({
+      hostname        = string
+      siteId          = string
+      systemIpAddress = string
+    })
+    additional_bandwidth = optional(number)
+  })
+  default = {
```

Comment:
We need proper validation for each field here for the secondary device. Otherwise, the default value will be used and bypass all the validation check.